src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.cpp

```diff
@@ -0,0 +1,86 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp"
+
+#include <array>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/LeviCivitaIterator.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace GeneralizedHarmonic::BoundaryConditions::detail {
+template <size_t VolumeDim, typename DataType>
+void set_dt_v_zero_constraint_preserving(
+    gsl::not_null<tnsr::iaa<DataType, VolumeDim, Frame::Inertial>*>
+        bc_dt_v_zero,
+    const tnsr::I<DataType, VolumeDim, Frame::Inertial>&
+        unit_interface_normal_vector,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>&
+        four_index_constraint,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_zero,
+    const std::array<DataType, 4>& char_speeds) noexcept {
+  ASSERT(get_size(get<0, 0, 0>(*bc_dt_v_zero)) ==
+             get_size(get<0>(unit_interface_normal_vector)),
+         "Size of input variables and temporary memory do not match: "
+             << get_size(get<0, 0, 0>(*bc_dt_v_zero)) << ","
+             << get_size(get<0>(unit_interface_normal_vector)));
+
+  for (size_t a = 0; a <= VolumeDim; ++a) {
+    for (size_t b = a; b <= VolumeDim; ++b) {
+      for (size_t i = 0; i < VolumeDim; ++i) {
+        bc_dt_v_zero->get(i, a, b) = char_projected_rhs_dt_v_zero.get(i, a, b);
+      }
+      // Lets say this term is T2_{kab} := - n_l N^l n^j C_{jkab}.
+      // But we store C4_{iab} = LeviCivita^{ijk} dphi_{jkab},
+      // which means  C_{jkab} = LeviCivita^{ijk} C4_{iab}
+      // where C4 is `four_index_constraint`.
+      // therefore, T2_{iab} =  char_speed<VZero> n^j C_{jiab}
+      // (since char_speed<VZero> = - n_l N^l), and therefore:
+      // T2_{iab} = char_speed<VZero> n^k LeviCivita^{ijk} C4_{jab}.
```

Comment:
Please keep the index of `n` consistent, either `n^j` (as above) or `n^k` as here

Also, I checked this vs the help text for the four index constraint (https://github.com/sxs-collaboration/spectre/blob/develop/src/Evolution/Systems/GeneralizedHarmonic/Constraints.hpp), which says `C_{kiab} = \epsilon^j_{ki} D_{jab}`. (The comment here uses `C4_{jab}` instead of `D_{jab}`.) Since `\epsilon^j_{ki}` is raised and lowered with the Kronecker delta, you can permute these indices to get the version written above

Reply:
changed indexing. Also renamed `C4` to `D`.